v3/src/models/data/category-set.ts

```diff
@@ -275,9 +275,7 @@ export const CategorySet = types.model("CategorySet", {
     self.invalidate()
   },
   setColorForCategory(value: string, color: string) {
-    if (self.index(value) != null) {
-      self.colors.set(value, color)
-    }
+    self.colors.set(value, color)
```

Comment:
I think you should special case the empty string here to remove a color from the map:
```suggestion
    if (color) {
      self.colors.set(value, color)
    } else {
      self.colors.delete(value)
    }
```